doc/users.md

```diff
@@ -1 +1,872 @@
 # Users Module
+Module | HTTP Method | URL | Description 
+--- | --- | --- | ---
+[Add User](#add) | POST | /add | Add User
+[Edit User](#edit) | PUT | /edit | Edit User
+[View User](#view) | GET | /view | View User
+[Delete User](#delete) | DELETE | /delete | Delete User
+
+## <a name="add"></a>Add Data User
+
+### Endpoint 
+POST /add
+
+### Database
+![](./user-layer.png)
+
+for add, you need get :
+- company_id, company_code, company_name from companies tables
+- branch_id, branch_code, branch_name from branches tables
+- region_id, region_code from regions tables
+- kios_id, kios_code from kios tables
+
+### Headers
+Key | Value 
+--- | ---
+Content-Type | application/json
+Accept | application/json
+
+### Request Payloads
+Name | Type | Example Value
+--- | --- | ---
+name | string | dennis  
+username | string | vdennis  
+password | string | cdcgi2020  
+email | string | vdennis@cdc.id  
+type_akses | string | 4  
+company_id | string | 5fb5df66-5f80-416d-a055-401bcf944ef7  
+region_id | string | 5fb5e66d-e47c-47fd-b4bf-4064cf944ef7  
+branch_id | string | 5fb5e437-69dc-4eb6-8f8c-4022cf944ef7  
+kios_id | string | 5fb5e626-65ac-473d-af2c-3ff2cf944ef7  
+```
+{
+    "name"       : "dennis",  
+    "username"   : "vdennis",  
+    "password"   : "cdcgi2020",  
+    "email"      : "vdennis@cdc.id", 
+    "type_akses" : "4",  
+    "company_id" : "5fb5df66-5f80-416d-a055-401bcf944ef7",
+    "region_id"  : "5fb5e66d-e47c-47fd-b4bf-4064cf944ef7",  
+    "branch_id"  : "5fb5e437-69dc-4eb6-8f8c-4022cf944ef7",  
+    "kios_id"    : "5fb5e626-65ac-473d-af2c-3ff2cf944ef7" 
+}
+```
+
+### Response Payloads
+HTTP Code | Status | Description
+--- | --- | ---
+400 | Bad Request | Bad request payload  
+500 | Internal Server Error | some un-handle error in server 
+201 | Created | Created
+```
+{
+    "status_code": "CDC-400",
+    "status_message": "Bad Request",
+    "data": null
+}
+```
+
+```
+{
+    "status_code": "CDC-201",
+    "status_message": "Created",
+    "data": {
+        "id":"1",
+        "name"         : "dennis",
+        "username"     : "vdennis",  
+        "password"     : "7f2994ecd8c1498cd0661589df5051f096fe1aca",  
+        "email"        : "vdennis@cdc.id", 
+        "type_akses"   : "4",  
+        "company_id"   : "5fb5df66-5f80-416d-a055-401bcf944ef7",
+        "company_code" : "WARI",
+        "company_name" : "PT. WAHANAARTHA RITELINDO",
+        "region_id"    : "5fb5e66d-e47c-47fd-b4bf-4064cf944ef7", 
+        "region_code"  : "WARI-03", 
+        "branch_id"    : "5fb5e437-69dc-4eb6-8f8c-4022cf944ef7",  
+        "branch_code"  : "77777",  
+        "branch_name"  : "WARI TEST 2",  
+        "kios_id"      : "5fb5e626-65ac-473d-af2c-3ff2cf944ef7", 
+        "kios_code"    : "77777",  
+        "created"      : "2020-12-02 09:03:44",
+        "modified"     : "2020-12-02 09:03:44"
+    }
+}
+```
+
+### Logic
+
```

Comment:
- please describe detail concept about type of access (type_akses) below
- explain the filter content in here. 
ex: 
if user login is superadmin in CDC, we need to define company_id, 
else need no company_id in payload request because we will grab from company_id of user login. 

this also applies to region_id, branch_id and kios_id.